### PR TITLE
[OSD-12647] update the operator to be able to set service orchestration

### PR DESCRIPTION
**Pull request by bmeng**
Update the API with the new serviceOrchestration related parameters
Update the main reconcile loop to handle the service orchestration after PD service created

Additional changes needed:
new configmap with pagerduty service orchestration rule created within operator namespace

### CODE REVIEW

**Comment by fahlmant** [comment]
@bmeng is there an OSD card for this? And some documentation on Service Orchestration?

**Comment by bmeng** [comment]
hi @fahlmant The card for this PR is in the title OSD-12647
this is the epic: https://issues.redhat.com/browse/SDE-2096 
this is the spike before the implementation: https://issues.redhat.com/browse/OSD-13250 
cc @NautiluX

File: pkg/pagerduty/service.go [hunk 1]
```diff
@@ -362,6 +378,56 @@ func (c *SvcClient) DisableService(data *Data) error {
 	return nil
 }
 
+// ToggleServiceOrchestration enables/disables the service orchestration for a given PD service
+func (c *SvcClient) ToggleServiceOrchestration(data *Data, active bool) error {
```
Comment (mjlshen) [inline pkg/pagerduty/service.go:382]: This is reasonable short-term, but should this be contributed upstream to https://github.com/PagerDuty/go-pagerduty/tree/main?

File: controllers/pagerdutyintegration/serviceorchestration_handler.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+		return err
+	}
+
+	if pdData.ServiceID == "" {
+		// PD service has not been created, skip the service orchestration steps
+		return nil
+	}
+	if pdData.ServiceOrchestrationState != "active" {
+		r.reqLogger.Info("enabling the service orchestration")
+		err := pdclient.ToggleServiceOrchestration(pdData, true)
```
Comment (mjlshen) [inline controllers/pagerdutyintegration/serviceorchestration_handler.go:79]: Wouldn't this enable ServiceOrchestration for every single PagerDuty service? (Is that intended? I may have forgotten)

File: api/v1alpha1/pagerdutyintegration_types.go [hunk 3]
```diff
@@ -50,6 +50,13 @@ type PagerDutyIntegrationSpec struct {
 
 	// Name and namespace in the target cluster where the secret is synced.
 	TargetSecretRef corev1.SecretReference `json:"targetSecretRef"`
+
+	ServiceOrchestration ServiceOrchestration `json:"serviceOrchestration"`
```
Comment (mjlshen) [inline api/v1alpha1/pagerdutyintegration_types.go:54]: Small nit, but can we add description comments to these new structs/fields?

**Review by mjlshen — COMMENTED** [review]
Looks good overall! Just a couple small comments/questions

File: controllers/pagerdutyintegration/serviceorchestration_handler.go (hunk 2 above)
Comment (bmeng) [reply controllers/pagerdutyintegration/serviceorchestration_handler.go:79]: In pagerdutyintegration_controller.go, this whole `handleServiceOrchestration()` function will be called only when `if pdi.Spec.ServiceOrchestration.Enabled`.
So the goal here is going to enable the service orchestration for all the osd/rosa clusters' PD service which match the selectors defined in the PagerdutyIntegration `osd`.

File: pkg/pagerduty/service.go (hunk 1 above)
Comment (bmeng) [reply pkg/pagerduty/service.go:382]: I love this idea. :100:  I will put it on my list.

File: controllers/pagerdutyintegration/serviceorchestration_handler.go (hunk 2 above)
Comment (mjlshen) [reply controllers/pagerdutyintegration/serviceorchestration_handler.go:79]: Right, thanks for clarifying - that makes much more sense!

File: api/v1alpha1/pagerdutyintegration_types.go (hunk 3 above)
Comment (bmeng) [reply api/v1alpha1/pagerdutyintegration_types.go:55]: Added, thanks!

File: pkg/pagerduty/service.go (hunk 1 above)
Comment (bmeng) [reply pkg/pagerduty/service.go:382]: https://issues.redhat.com/browse/OSD-13926 created here.

**Comment by mjlshen** [comment]
Looks good to me, to pass the validate PR check I think you'll need to rebase and `make container-generate`

**Review by mjlshen — APPROVED** [review]
/lgtm

**Pull request merged**